abci/client/client.go

```diff
@@ -36,6 +36,11 @@ type Client interface {
 	// with the exception of `CheckTxAsync` which we maintain
 	// for the v0 mempool. We should explore refactoring the
 	// mempool to remove this vestige behavior.
+	//
+	// SetResponseCallback is not used anymore. The callback was invoked only by the mempool on
+	// CheckTx responses, only during rechecking. Now the responses are handled by the callback of
+	// the *ReqRes struct returned by CheckTxAsync. This callback is more flexible as it allows to
+	// pass other information such as the sender.
 	SetResponseCallback(cb Callback)
```

Comment:
I just checked, and:

* This no longer used anywhere in CometBFT (except for a proxy class, which is in turn not used
* On `main` branch of the SDK, they don't use it for anything.

Since CometBFT `v1.x` will be a major version. I think we could even remove it in `v1.x`. What do you think? If you find it to reckless, we can deprecate in on `v1.x`, as @melekes suggests, and remove it altogether on `main`.

Reply:
I marked it as deprecated.